internal/build/gen_mk.sh

```diff
@@ -147,78 +146,51 @@ gen_make_targets() {
 		sequential=false ;;
 	esac
 
-		# cd $dir
-		if [ "." = "$dir" ]; then
-			# root
-			cd=
-		else
-			cd="cd '$dir'; "
-		fi
+	# cd $dir
+	if [ "." = "$dir" ]; then
+		# root
+		cd=
+	else
+		cd="cd '$dir'; "
+	fi
 
-		if [ "$name" = root ]; then
-			# special case
-			case "$cmd" in
-			get)
-				cmdx="get -tags tools"
-				;;
-			up)
-				cmdx="get -tags tools -u"
-				;;
-			*)
-				cmdx=
-				;;
-			esac
-
-			[ -z "$cmdx" ] || cmdx="\$(GO) $cmdx -v ./..."
-
-			case "$cmd" in
-			up)
-				callx="$cmdx
-\$(GO) mod tidy"
-				;;
-			get)
-				callx="$cmdx"
-				;;
-			*)
-				callx="$call"
-				;;
-			esac
-		else
-			callx="$call"
+	case "$cmd" in
+	build)
+		# special build flags for cmd/*
+		#
+		callx="$(cat <<-EOL | packed_oneline
+		set -e
+		MOD="\$\$(\$(GO) list -f '{{.ImportPath}}' ./...)"
+		if echo "\$\$MOD" | grep -q -e '.*/cmd/[^/]\+\$\$'; then
+			\$(GO_BUILD_CMD) ./...
+		elif [ -n "\$\$MOD" ]; then
+			\$(GO_BUILD) ./...
 		fi
-
-		if [ "build" = "$cmd" ]; then
-			# special build flags for cmd/*
-			#
-			callx="$(cat <<-EOL | packed_oneline
-			set -e
-			MOD="\$\$(\$(GO) list -f '{{.ImportPath}}' ./...)"
-			if echo "\$\$MOD" | grep -q -e '.*/cmd/[^/]\+\$\$'; then
-				\$(GO_BUILD_CMD) ./...
-			elif [ -n "\$\$MOD" ]; then
-				\$(GO_BUILD) ./...
-			fi
-			EOL
-			)"
+		EOL
```

Comment:
interesting, we would need to test if this work when `./cmd/foo/` contains a `go.mod`

Reply:
Interesting that rabbit acknowledged a potential issue but did not propose a solution

Reply:
> Interesting that rabbit acknowledged a potential issue but did not propose a solution 

the proposed fix is nice for inspiration, but regardless the nested go.mod issue it's is buggy as it would build cmd/foo twice.